internal/audit/otel-audit.go

```diff
@@ -45,14 +45,19 @@ func (c *AuditClient) Send(ctx context.Context, msg msgs.Msg, options ...base.Se
 	return c.client.Send(ctx, msg)
 }
 
-func NewOtelAuditClient(remoteAddress string, options ...base.Option) (*AuditClient, error) {
+func NewOtelAuditClient(connectSocket bool, remoteAddress string, options ...base.Option) (*AuditClient, error) {
```

Comment:
This is an awkward function IMO - just let the user pass in the connection, passing in proto-parameters and doing a `switch` saves nothing

Reply:
Moved the switch part to a separate function and passed the audit.CreateConn to this function.